pages/_document.tsx

```diff
@@ -0,0 +1,21 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
+
+class MyDocument extends Document {
+  render() {
+    return (
+      <Html>
+        <Head>
```

Comment:
I would suggest having self-hosted font-family instead of having the Google's API. Here are some articles:

[https://blog.logrocket.com/self-hosted-fonts-vs-google-fonts-api/](https://blog.logrocket.com/self-hosted-fonts-vs-google-fonts-api/)
[https://www.tunetheweb.com/blog/should-you-self-host-google-fonts/](https://www.tunetheweb.com/blog/should-you-self-host-google-fonts/)


Maybe, also, you could create a file and have the font-face there and then imported to the _document file. For example:

import React, { Fragment } from "react";

const WebFonts = () => (
  <Fragment>
    <style global jsx>{`
      @font-face {
        font-family: "Source Sans Pro";
        font-style: normal;
        font-stretch: normal;
        font-weight: 400;
        font-display: fallback;
        src: local("SourceSansPro Regular"), local("SourceSansPro-Regular"),
          url(/static/fonts/SourceSansPro-Regular.woff2) format("woff2");
        unicode-range: U+0100-024f, U+1-1eff, U+20a0-20ab, U+20ad-20cf, U+2c60-2c7f,
          U+A720-A7FF;
      }
      @font-face {
        font-family: "Source Sans Pro";
        font-style: normal;
        font-weight: 700;
        font-display: fallback;
        src: local("SourceSansPro SemiBold"), local("SourceSansPro-SemiBold"),
          url(/static/fonts/SourceSansPro-Bold.woff2) format("woff2");
        unicode-range: U+0100-024f, U+1-1eff, U+20a0-20ab, U+20ad-20cf, U+2c60-2c7f,
          U+A720-A7FF;
      }
    `}</style>
  </Fragment>
);

export default WebFonts;